schema-nft-marketplace.graphql

```diff
@@ -0,0 +1,255 @@
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenue and creatorRevenue. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the collection on the marketplace. "
+    tradeCount: Int!
+
+    " Trades of the collection. "
+    trades: [Trade!]! @derivedFrom(field: "collection")
+}
+
+" Trades exist such as a combination of taker/order and bid/ask. "
+type Trade @entity {
+    " { maker txn hash }-{ taker txn hash } "
+    id: ID!
+
+    " Transaction hash of the maker. "
+    makerTransactionHash: String!
+
+    " Transaction hash of the taker. "
+    takerTransactionHash: String!
+
+    " Event log index. "
+    logIndex: Int!
+
+    " Block timestamp where the trade is executed. "
+    timestamp: BigInt!
+
+    " Block number where the trade is executed. "
+    blockNumber: BigInt!
+
+    " Collection involved "
+    collection: Collection!
+
+    " Token ID of the traded NFT. "
+    tokenId: BigInt!
+
+    " Price (in ETH) "
+    priceETH: BigDecimal!
+
+    " Stretegy that the trade is executed. "
+    strategy: SaleStrategy!
+
+    " Buyer account address "
+    buyer: String!
+
+    " Seller account address "
+    seller: String!
+}
+
+type MarketplaceDailySnapshot @entity {
+    " { Contract address }-{# of days since Unix epoch time} "
+    id: ID!
+
+    " Block number where the snapshot is taken. "
+    blockNumber: BigInt!
+
+    " Block timestamp when the snapshot is taken. "
+    timestamp: BigInt!
+
+    " Number of collections listed on the marketplace. "
+    collectionCount: Int!
+
+    " Cumulative trade volume (in ETH) "
+    cumulativeTradeVolumeETH: BigDecimal!
+
+    " Revenue that goes to the marketplace protocol, aka protocol fee. "
+    marketplaceRevenueETH: BigDecimal!
+
+    " Revenue that goes to creator, aka royalty fee. "
+    creatorRevenueETH: BigDecimal!
+
+    " Sum of marketplaceRevenueETH and creatorRevenueETH. "
+    totalRevenueETH: BigDecimal!
+
+    " Trade count of the all collections on the marketplace. "
+    tradeCount: Int!
+
+    " Cumulative unique traders "
+    cumulativeUniqueTraders: Int!
+
+    " Daily active traders "
+    dailyActiveTraders: Int!
+
+    " Number of traded collections of the day "
+    dailyTradedCollectionCount: Int!
+
+    " Number of traded items of the day "
+    dailyTradedItemCount: Int!
+}
+
+type CollectionDailySnapshot @entity {
+    " { Contract address }-{# of days since epoch unix time } "
+    id: ID!
+
+    " Block number where the snapshot is taken. "
+    blockNumber: BigInt!
+
+    " Block timestamp when the snapshot is taken. "
+    timestamp: BigInt!
+
+    " Royalty fee rate in percentage. E.g. 2.5% should be 2.5 "
+    royaltyFee: BigDecimal!
+
+    " Number of tokens listed for sale in the marketplace. "
+    listedCount: BigInt!
```

Comment:
why don't we add the following fields?
```
   "  # of unique users holding this collection/NFT "
    holdersCount: Int!

   "  # of unique users buying this collection/NFT "
    buyersCount: Int!

   " # of unique users selling this collection/NFT "
    sellersCount: Int!
```

Reply:
We will do holder in a separate subgraph. But it does make sense to separate buyers and sellers